src/adapters/EasyRtcAdapter.js

```diff
@@ -173,58 +183,114 @@ class EasyRtcAdapter extends NoOpAdapter {
+    clientMediaStreams.set(streamName, stream);
 
-    this.mediaStreams[clientId] = { audio: audioStream, video: videoStream };
-
-    // Resolve the promise for the user's media stream if it exists.
     if (this.pendingMediaRequests.has(clientId)) {
-      this.pendingMediaRequests.get(clientId).audio.resolve(audioStream);
-      this.pendingMediaRequests.get(clientId).video.resolve(videoStream);
+      const pendingMediaRequests = this.pendingMediaRequests.get(clientId);
+
+      // Add mediaStreams audio streamName alias if does not exist yet
+      const audioTracks = stream.getAudioTracks();
+      if (!clientMediaStreams.has('audio') && audioTracks.length > 0) {
+        // Safari doesn't like it when you use single a mixed media stream where one of the tracks is inactive, so we
+        // split the tracks into two streams.
+        const audioStream = new MediaStream();
+        try {
+          audioTracks.forEach(track => audioStream.addTrack(track));
+          clientMediaStreams.set('audio', audioStream)
+        } catch(e) {
+          NAF.log.warn(`${clientId} setMediaStream "audio" alias Error`, e);
+        }
+
+        // Resolve the promise for the user's media stream audio alias if it exists.
+        pendingMediaRequests.audio.resolve(audioStream);
+      }
+
+      // Add mediaStreams video streamName alias if does not exist yet
+      const videoTracks = stream.getVideoTracks();
+      if (!clientMediaStreams.has('video') && videoTracks.length > 0) {
+        // Safari doesn't like it when you use single a mixed media stream where one of the tracks is inactive, so we
+        // split the tracks into two streams.
+        const videoStream = new MediaStream();
+        try {
+          videoTracks.forEach(track => videoStream.addTrack(track));
+          clientMediaStreams.set('video', videoStream)
+        } catch(e) {
+          NAF.log.warn(`${clientId} setMediaStream "video" alias Error`, e);
+        }
+
+        // Resolve the promise for the user's media stream video alias if it exists.
+        pendingMediaRequests.video.resolve(videoStream);
+      }
+
+      // Resolve the promise for the user's media stream by StreamName if it exists.
+      if (pendingMediaRequests[streamName]) {
+        pendingMediaRequests[streamName].resolve(stream);
+      }
     }
   }
 
+  addLocalMediaStream(stream, streamName) {
+    const easyrtc = this.easyrtc;
+    streamName = streamName || stream.id;
+    this.setMediaStream("local", stream, streamName);
+    easyrtc.register3rdPartyLocalMediaStream(stream, streamName);
+
+    // Add local stream to existing connections
+    Object.keys(this.remoteClients).forEach((clientId) => {
+      if (easyrtc.getConnectStatus(clientId) !== easyrtc.NOT_CONNECTED) {
+        easyrtc.addStreamToCall(clientId, streamName);
+      }
+    });
+  }
+
+  removeLocalMediaStream(stream, streamName) {
```

Comment:
I would have simplified it to just `removeLocalMediaStream(streamName)` Wouldn't it be better? This way you don't need to keep the stream object in a variable after you call `addLocalMediaStrem` to call `removeLocalMediaStream` later.